server.js

```diff
@@ -20,4 +26,53 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+//send messages
+
+app.post("/messages", function (request, response) {
+  let newMessage = request.body;
+  if (newMessage.from === "" || newMessage.text === "") {
+    throw new Error("400");
+  } else {
+    messages.push(newMessage);
+    response.json(newMessage);
+    console.log(newMessage);
+  }
+});
+
+//read all messages
+
+app.get("/messages", function (request, response) {
+  response.json(messages);
+});
+
+app.get("/messages/:id", function (request, response) {
+  const idToFind = Number(request.params.id);
+  const message = messages.find((message) => message.id === idToFind);
+  console.log(message);
+  response.json(message);
+});
+
+//delete  messages
+
+app.delete("/messages/:id", function (request, response) {
+  const idToFind = Number(request.params.id);
+  const message = messages.find((message) => message.id === idToFind);
+  console.log(message);
+  response.json(message);
+});
+
+//latest messages
+
+app.get("/messages/latest", (req, res) => {
+  let result = messages.slice(-10);
+  res.json(result);
+});
+
+app.get("/messages/search", (req, res) => {
+  let result = messages.filter((msg) =>
+    msg.text.toLowerCase().includes(req.query.text)
+  );
+  res.send(result);
+});
+
```

Comment:
its good to have some msgs for app.linsten to make sure when you run it, its working
```node
const port = 8080;
app.listen(port, () => {
  console.log(`Server is running on http://localhost:${port}`);
});
```